### ISSUE TITLE
null list input turns into [null] instead of null

### ISSUE THREAD

**Issue by amay** [graphql-dotnet/conventions#73]
I'm running into some weird behavior when attempting to work with a list of inputs that I want to be able to be nullable.

If I send the following query:

```json
{
  "operationName":"TestQuery",
  "query":"query TestQuery($testInputs: [TestInput]) {\n  example(testInputs: $testInputs)\n}\n",
  "variables":{"testInputs":null}
}
```

With the query defined as:

```csharp
using System.Collections.Generic;
using System.Linq;
using GraphQL.Conventions;

namespace Test.GraphQL.Schema
{
    public class TestQuery
    {
        public string Example(IEnumerable<TestInput> testInputs)
        {
            return testInputsString = testInputs == null
                ? "null"
                : "[" + string.Join(",", testInputs.Select(x => x == null ? "null" : x.Text)) + "]";
        }
    }

    [InputType]
    public class TestInput
    {
        public string Text { get; set; }
    }
}
```

I get the following return value:

```json
{"data":{"example":"[null]"}}
```

But I'd expect to get:

```json
{"data":{"example":"null"}}
```

However, I do get the expected result if I run the query without the variable:
```json
{
  "query":"{\n  example\n}\n",
  "variables":{}
}
```

I'm fairly new to GraphQL, so this could very likely be me misunderstanding how the interplay between queries and variables works, but the list having the value of `[null]` seems very surprising to me. Any help tracking this down is greatly appreciated!

**Comment by tlil** [comment]
Interesting – thanks for reporting – I will try to rebase off of latest `master` in [graphql-dotnet/graphql-dotnet](https://github.com/graphql-dotnet/graphql-dotnet) and see if that fixes it – I pulled in the `null` literal support from the work there before it made it to the official release, so might be a result of that. 

And sorry for the late reply. I've been out for a month for my honeymoon. Will try to take a look at this in the next few days.

**Comment by amay** [comment]
Hey @tlil, thanks for the response and congrats!

Any luck reproducing this issue?

**Comment by tlil** [comment]
Hey @amay - yes, I did. I found the issue here now. Will push a new version in a few minutes. Thanks for reporting!

**Comment by amay** [comment]
Thanks for the fix!

**Comment by tlil** [comment]
No probs. FYI @amay - long overdue, but I've now raised a PR against graphql-dotnet/graphql-dotnet as well to fix this in #82
